CRM/ACL/Page/ACL.php

```diff
@@ -185,7 +185,12 @@ public function browse() {
       $acl[$dao->id]['is_active'] = $dao->is_active;
 
       if ($acl[$dao->id]['entity_id']) {
-        $acl[$dao->id]['entity'] = $roles[$acl[$dao->id]['entity_id']];
+        // take disabled roles into account
+        if (isset($roles[$acl[$dao->id]['entity_id']])) {
+           $acl[$dao->id]['entity'] = $roles[$acl[$dao->id]['entity_id']];
+        } else {
+           $acl[$dao->id]['entity'] = "";
+        }
```

Comment:
There is util function which does the same job also you can provide default value on empty. Like here 
`$acl[$dao->id]['entity'] = CRM_Utils_Array::value($acl[$dao->id]['entity'], $roles, '');
`